.github/workflows/maintainer_bors.yml

```diff
@@ -0,0 +1,93 @@
+    runs-on: ubuntu-latest
+    steps:
+      - name: Find bors merge/delegate
+        id: merge_or_delegate
+        run: |
+          COMMENT="${COMMENT_EVENT}${COMMENT_REVIEW}"
+          # we strip `\r` since line endings from GitHub contain this character
+          COMMENT="${COMMENT//$'\r'/}"
+          # for debugging, we print some information
+          printf '%s' "${COMMENT}" | hexdump -cC
+          printf 'Comment:"%s"\n' "${COMMENT}"
+          m_or_d="$(printf '%s' "${COMMENT}" |
+            sed -n 's=^bors  *\(merge\|r+\)$=ready-to-merge=p; s=^bors  *d.*=delegated=p' | head -1)"
+
+          printf $'"bors delegate" or "bors merge" found? \'%s\'\n' "${m_or_d}"
+
+          printf $'mOrD=%s\n' "${m_or_d}" > "${GITHUB_OUTPUT}"
+          if [ "${AUTHOR}" == 'leanprover-community-mathlib4-bot' ] ||
+             [ "${AUTHOR}" == 'leanprover-community-bot-assistant' ]
+          then
+            printf $'bot=true\n'
+            printf $'bot=true\n' > "${GITHUB_OUTPUT}"
+          else
+            printf $'bot=false\n'
+            printf $'bot=false\n' > "${GITHUB_OUTPUT}"
+          fi
+
+      - name: Check whether user is a mathlib admin
+        id: user_permission
+        if: ${{ ! steps.merge_or_delegate.outputs.mOrD == '' }}
+        uses: actions-cool/check-user-permission@v2
+        with:
+          require: 'admin'
+          # review(_comment) use
+          # require: 'write'
+          # token: ${{ secrets.TRIAGE_TOKEN }}
+
+      - name: Add ready-to-merge or delegated label
+        id: add_label
+        if: ${{ ! steps.merge_or_delegate.outputs.mOrD == '' &&
+          ( steps.user_permission.outputs.require-result == 'true' ||
+            steps.merge_or_delegate.outputs.bot == 'true' ) }}
+        uses: octokit/request-action@v2.x
+        with:
+          # check is this ok? was /repos/:repository/issues/:issue_number/labels
+          route: POST /repos/:repository/issues/${PR_NUMBER}/labels
+          repository: ${{ github.repository }}
+          issue_number: ${PR_NUMBER}
+          labels: '["${{ steps.merge_or_delegate.outputs.mOrD }}"]'
+        env:
+          GITHUB_TOKEN: ${{ secrets.TRIAGE_TOKEN }} # comment uses ${{ secrets.GITHUB_TOKEN }}
+
+      - if: ${{ ! steps.merge_or_delegate.outputs.mOrD == '' &&
+          ( steps.user_permission.outputs.require-result == 'true' ||
+            steps.merge_or_delegate.outputs.bot == 'true' ) }}
+        id: remove_labels
+        name: Remove "awaiting-author"
+        # we use curl rather than octokit/request-action so that the job won't fail
+        # (and send an annoying email) if the labels don't exist
+        run: |
+          curl --request DELETE \
+            --url "https://api.github.com/repos/${{ github.repository }}/issues/${PR_NUMBER}/labels/awaiting-author" \
+            --header 'authorization: Bearer ${{ secrets.TRIAGE_TOKEN }}'
+            # comment uses ${{ secrets.GITHUB_TOKEN }}
```

Comment:
IDem

Reply:
Same situation as above: the `comment` action uses `GITHUB_TOKEN`, but I preferred `TRIAGE` due to GH quota considerations.